be/src/exprs/agg/window.h

```diff
@@ -412,9 +422,23 @@ template <LogicalType LT, bool ignoreNulls, bool isLag, typename T = RunTimeCppT
 class LeadLagWindowFunction final : public ValueWindowFunction<LT, LeadLagState<LT>, T> {
     using InputColumnType = typename ValueWindowFunction<LT, FirstValueState<LT>, T>::InputColumnType;
 
+    mutable int64_t offset = 0;
```

Comment:
Could `offset` stored be in `LeadLagState` instead of a field of `LeadLagWindowFunction`

Reply:
Yes, but unnecessary, it will take more space.